### PR TITLE
[RFR] Better return fields

### PR DESCRIPTION
**Pull request by ThieryMichel**
follow #25

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ test:
   volumes:
     - .:/app
   working_dir: /app
-  command: node_modules/mocha/bin/mocha --harmony test/setup.js test/**/*.js
+  command: node_modules/mocha/bin/mocha --harmony test/setup.js test/**/*.js ./lib/**/*.spec.js
```
Comment (jpetitcolas) [inline docker-compose.yml]: We encountered some troubles with globs. I would recommend using following syntax instead:

```
NODE_ENV=test ./node_modules/.bin/mocha \
		./tests/setup.js \
		$$(find tests/e2e -name '*.spec.js')
```

File: lib/queries/batchInsert.js [hunk 2]
```diff
@@ -1,42 +1,25 @@
...
-        table,
-        fields,
-    } = this.config;
+export default ({ table, fields, returnFields }) => {
+    const returning = returningQuery(returnFields);
 
-    let { returnFields } = this.config;
+    return (entities) => {
+        if (!entities || entities.length === 0) {
+            throw new Error(`No data for batch inserting ${table} entities.`);
```
Comment (jpetitcolas) [inline lib/queries/batchInsert.js]: Not sure we need to throw an error here. Batch inserting nothing should work fine (even if it does nothing).

File: docker-compose.yml (hunk 1 above)
Comment (Kmaschta) [inline docker-compose.yml]: Globs works just fine but you need to put in in quotes too (e.g. `"test/**/*.js"`)

File: lib/queries/batchInsert.js [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
 
-    const parameters = batchParameter(fields)(entities);
-    const getValueSubQuery = valueSubQuery(fields);
+export default ({ table, fields, returnFields }) => {
+    const returning = returningQuery(returnFields);
 
-    const returning = returnFields.join(', ');
+    return (entities) => {
+        if (!entities || entities.length === 0) {
+            return { sql: '' };
```
Comment (djhi) [inline lib/queries/batchInsert.js]: empty string ? How is it handled later ?

Comment (ThieryMichel) [inline lib/queries/batchInsert.js]: Sql will do nothing, but I realize it is not optimal.

Comment (djhi) [inline lib/queries/batchInsert.js]: An additional check in the client might be nice to avoid running empty string queries ?

Comment (ThieryMichel) [inline lib/queries/batchInsert.js]: done ^^

**Pull request merged**